File: src/beginner_tutorials/src/scripts/iot_listener.py
#!/usr/bin/env python
import rospy
from beginner_tutorials.msg import IoTSensor

def get_sensor_values(message):
    sensor_id = message.id
    sensor_name = message.name
    temperature = message.temperature
    humidity    = message.humidity
    rospy.loginfo(" %s (%d) - Temperature: %.2f Humidity: %.2f"%(
        sensor_name, sensor_id, temperature, humidity))

def listener():
    rospy.init_node("iot_listener")
    rospy.loginfo("IoT Listener initialized")
    rospy.Subscriber("iot_line", IoTSensor, get_sensor_values)
    rospy.spin()

if __name__ == "__main__":
    listener()

File: src/beginner_tutorials/src/iot_listener.cpp
#include "ros/ros.h"
#include "beginner_tutorials/IoTSensor.h"

void iot_sensor_callback(const beginner_tutorials::IoTSensor::ConstPtr& msg);

int main(int argc, char **argv){
    ros::init(argc, argv, "iot_listener");
    ros::NodeHandle n;
    ros::Subscriber sub = n.subscribe("iot_line", 1000, iot_sensor_callback);
    ros::spin();
    return 0;
}

void iot_sensor_callback(const beginner_tutorials::IoTSensor::ConstPtr& msg){
    ROS_INFO("[IoTSensor::%s (%d)] - Temperature: %.2f, Humidity: %.2f",
        msg->name.c_str(), msg->id, msg->temperature, msg->humidity);
}
